Core/Core/SwiftUIViews/ListCellView/ListCellView.swift

```diff
@@ -0,0 +1,263 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+struct ListCellView: View {
+
+    enum ListCellStyle {
+        case mainAccordionHeader
+        case listAccordionHeader
+        case listItem
+    }
+
+    enum SelectionState {
+        case deselected
+        case selected
+        case partiallySelected
+    }
+
+    @ObservedObject var viewModel: ListCellViewModel
+
+    internal init(_ viewModel: ListCellViewModel) {
+        self.viewModel = viewModel
+    }
+
+    @ViewBuilder
+    private var iconImage: some View {
+        if viewModel.selectionDidToggle != nil {
+            HStack {
+                switch viewModel.selectionState {
+                case .deselected:
+                    Image.emptyLine
+                        .size(20)
+                        .foregroundColor(.textDarkest)
+                case .selected:
+                    Image.completeSolid
+                        .size(20)
+                        .foregroundColor(.textInfo)
+                case .partiallySelected:
+                    Image.partialSolid
+                        .size(20)
+                        .foregroundColor(.textInfo)
+                }
+            }.frame(width: 32, height: 32)
+                .padding(.leading, 24)
+                .padding(.trailing, 4)
+        }
+    }
+
+    @ViewBuilder
+    private var accessoryIcon: some View {
+        HStack(spacing: 0) {
+            switch viewModel.cellStyle {
+            case .mainAccordionHeader, .listAccordionHeader:
+                if let isCollapsed = viewModel.isCollapsed {
+                    Image.arrowOpenDownLine
+                        .size(24)
+                        .foregroundColor(.textDarkest)
+                        .rotationEffect(isCollapsed ? .degrees(0) : .degrees(-180))
+                }
+            case .listItem:
+                SwiftUI.EmptyView()
+            }
+            progressAccessoryView
+        }
+        .padding(16)
+    }
+
+    @ViewBuilder
+    private var progressAccessoryView: some View {
+            if viewModel.error != nil {
```

Comment:
The indentation of the whole function is off.

Reply:
fixed.